programs/gauge/src/state.rs

```diff
@@ -193,25 +196,24 @@ impl GaugeVote {
         return Err(VotingEpochNotFound.into());
     }
 
-    pub fn reset_voting_epoch(&mut self, current_vote_index: usize) {
+    pub fn reset_voting_epoch(&mut self, current_vote_index: usize) -> Result<()> {
         self.vote_epochs[current_vote_index] = GaugeVoteItem::default();
-        let current_index = current_vote_index
-            .checked_add(MAX_EPOCH_PER_GAUGE)
-            .unwrap()
-            .checked_sub(1)
-            .unwrap()
-            % MAX_EPOCH_PER_GAUGE;
-        self.current_index = u64::try_from(current_index).unwrap();
+        let current_index = current_vote_index.safe_add(MAX_EPOCH_PER_GAUGE)?;
+        let current_index = current_index.safe_sub(1)? % MAX_EPOCH_PER_GAUGE;
+
+        self.current_index =
+            u64::try_from(current_index).map_err(|_e| VipersError::IntegerOverflow)?;
+        Ok(())
     }
 
     pub fn claim_a_fee(&mut self, to_epoch: u32, gauge: &Gauge) -> Result<u64> {
         let mut total_fee_amount = 0u64;
         for i in self.last_claim_a_fee_epoch..=to_epoch {
             match self.get_index_for_voting_epoch(i) {
                 Ok(index) => {
-                    let fee_amount = gauge
-                        .get_allocated_fee_a(self.vote_epochs[index].allocated_power, i)
-                        .unwrap();
+                    let fee_amount = unwrap_opt!(
```

Comment:
we should use or_or_else

Reply:
since `get_allocated_fee_a` function is updated to return `Result`, we can propagate error here.